osgar/drivers/rosmsg.py

```diff
@@ -158,16 +159,14 @@ def parse_laser(data):
 
     # conversion to int and millimeters
     scan = [int(x*1000) if x < 65.0 else 0 for x in scan]
-    #scan = [0] * 640 + scan + [0] * 640
-    to_cut = int(len(scan) / 360 * 45)  #original scan is 360deg
-    scan = scan[to_cut:len(scan)-to_cut]
-
 
-    #reduced_scan = []
-    #for index,value in enumerate(scan):
-    #    if index % 10 == 1:
-    #        reduced_scan.append(value)
+    if angle_range_deg == 270:
+        # SubT Virtual World
+        return scan
 
+    # scan from map for MOBoS, Maria, K2, ...
+    to_cut = int(len(scan) / 360 * 45)  # original scan is 360deg
+    scan = scan[to_cut:len(scan)-to_cut]
```

Comment:
I haven't realized that the scans from local map are 360 degrees. That would make Mobos and Maria symetric which would allow us to back up while following wall. In the future, I say we should go for it.